src/apint/arithmetic.rs

```diff
@@ -919,68 +1791,374 @@ mod tests {
         }
     }
 
-    mod udiv {
-        use super::*;
-
-        #[test]
-        fn simple() {
-            let lhs = ApInt::from(56_u32);
-            let rhs = ApInt::from(7_u32);
-            let result = lhs.into_wrapping_udiv(&rhs).unwrap();
-            assert_eq!(result, ApInt::from(8_u32));
-        }
-    }
-
-    mod sdiv {
+    mod div_rem {
         use super::*;
+        use bitwidth::BitWidth;
+        use std::u64;
 
+        //TODO: add division by zero testing after error refactoring is finished
+        //use errors::ErrorKind;
         #[test]
         fn simple() {
-            let lhs = ApInt::from(72_i32);
-            let rhs = ApInt::from(12_i32);
-            let result = lhs.into_wrapping_sdiv(&rhs).unwrap();
-            assert_eq!(result, ApInt::from(6_u32));
+            macro_rules! s {
```

Comment:
Please add some documentation to this macro (even if its test code) and maybe restructure the parameters so they are more readable. Like columnwise.

Reply:
Also I am not sure if you really need a macro here at all. A simple (generic) function should suffice.

Reply:
When I add the error checking and potentially other things, this macro will really save on lines and effort. I think it is already worth it and in fact it helped me find a bug.